### ISSUE TITLE
questions about getting test scores

### ISSUE THREAD

**Issue by jessie-chen99** [WuJunde/MedSegDiff#34]
Are the scores in tableⅠ of MedSedDiff the official test scores or 5-fold cross validation scores? nii.gz files are needed to be uploaded to BraTS2020 website to get the official test score, but I cannot find nii.gz generating section in related source code so I don't know how to get the test score of trained model.

**Comment by lqy9715** [comment]
Hi, I've also encountered a problem in this area, how do I perform, for example, a dice test after generating an image? Do I need to write a test.py program?

**Comment by jessie-chen99** [comment]
> Hi, I've also encountered a problem in this area, how do I perform, for example, a dice test after generating an image? Do I need to write a test.py program?

I just add a dice score calculate function after generating an image in sample.py. I noticed that this V1-version code is a 2-class segmentation task instead of 4-class(official), thus the official testing is not available. Maybe V2-version code will fix that.

**Comment by lqy9715** [comment]
Thanks for your reply and one more question to seek your advice. When the hyperparameter - num_ensemble 5 - samples out 5 images, how do you select the images among them or what is the strategy to select these images to calculate dice with ground truth.

**Comment by jaceqin** [comment]
> > 嗨，我在这方面也遇到了一个问题，例如，生成图像后如何执行骰子测试？我需要编写 test.py 程序吗？
> 
> 我注意到这个 V1 版本的代码是 2 类分段任务而不是 4 类（官方），因此官方测试不可用。所以我只是在 _segmentation_sample_.py 中生成图像后添加一个骰子分数计算函数。也许 V2 版本的代码可以解决这个问题。

Can you share your code for calculating dice score？

**Comment by jaceqin** [comment]
> Are the scores in tableⅠ of MedSedDiff the official test scores or 5-fold cross validation scores? nii.gz files are needed to be uploaded to BraTS2020 website to get the official test score, but I cannot find nii.gz generating section in related source code so I don't know how to get the test score of trained model.

I am a beginner,Can you share your code for calculating dice score？Thank you!

**Comment by lqy9715** [comment]
> > Are the scores in tableⅠ of MedSedDiff the official test scores or 5-fold cross validation scores? nii.gz files are needed to be uploaded to BraTS2020 website to get the official test score, but I cannot find nii.gz generating section in related source code so I don't know how to get the test score of trained model.
> 
> I am a beginner,Can you share your code for calculating dice score？Thank you!

Change the dataset loader code a bit to load the test set's mask, so that dice are calculated after the sample.

**Comment by jaceqin** [comment]
> > > Are the scores in tableⅠ of MedSedDiff the official test scores or 5-fold cross validation scores? nii.gz files are needed to be uploaded to BraTS2020 website to get the official test score, but I cannot find nii.gz generating section in related source code so I don't know how to get the test score of trained model.
> > 
> > 
> > I am a beginner,Can you share your code for calculating dice score？Thank you!
> 
> Change the dataset loader code a bit to load the test set's mask, so that dice are calculated after the sample.

我试了好几次但是都不行，可以加你的联系方式请教你一下吗？

**Comment by jaceqin** [comment]
> > > Are the scores in tableⅠ of MedSedDiff the official test scores or 5-fold cross validation scores? nii.gz files are needed to be uploaded to BraTS2020 website to get the official test score, but I cannot find nii.gz generating section in related source code so I don't know how to get the test score of trained model.
> > 
> > 
> > I am a beginner,Can you share your code for calculating dice score？Thank you!
> 
> Change the dataset loader code a bit to load the test set's mask, so that dice are calculated after the sample.

可以看下代码吗？还没解决

**Comment by cyz21** [comment]
> > > Are the scores in tableⅠ of MedSedDiff the official test scores or 5-fold cross validation scores? nii.gz files are needed to be uploaded to BraTS2020 website to get the official test score, but I cannot find nii.gz generating section in related source code so I don't know how to get the test score of trained model.
> > 
> > 
> > I am a beginner,Can you share your code for calculating dice score？Thank you!
> 
> Change the dataset loader code a bit to load the test set's mask, so that dice are calculated after the sample.

Hello, I also tried many unsuccessful. How to modify the code to get the dice score? Could you please share or elaborate? Thank you very much. If you would like to share with me, that would be great.Thank you again!
